src/plugins/github/graphql.test.js

```diff
@@ -631,4 +637,331 @@ describe("graphql", () => {
+        "reviewComments",
+        "reviews",
+        "whoami",
+      ]);
+      result.forEach((fd) => expect(createFragments()).toContainEqual(fd));
+    });
+  });
+
+  describe("#postQueryExhaustive", () => {
+    it("resolves a representative query", async () => {
+      const makeAuthor = (name) => ({
+        __typename: "User",
+        login: name,
+        id: `opaque-user-${name}`,
+      });
+      // We'll have three stages:
+      //   - The original result will need more issues, and more
+      //     comments for issue 1, and more reviews for PR 2.
+      //   - The next result will need more issues, and comments for
+      //     issues 1 (original issue) and 3 (new issue).
+      //   - The final result will need no more data.
+      // We obey the contract pretty much exactly, except that we return
+      // far fewer results than are asked for by the query.
+      //
+      // Here is the response to the initial query.
+      const response0 = {
+        repository: {
+          id: "opaque-repo",
+          issues: {
+            pageInfo: {
+              hasNextPage: true,
+              endCursor: "opaque-cursor-issues-v0",
+            },
+            nodes: [
+              {
+                id: "opaque-issue1",
+                title: "Request for comments",
+                body: "Like it says, please comment!",
+                number: 1,
+                author: makeAuthor("decentralion"),
+                comments: {
+                  pageInfo: {
+                    hasNextPage: true,
+                    endCursor: "opaque-cursor-issue1comments-v0",
+                  },
+                  nodes: [
+                    {
+                      id: "opaque-issue1comment1",
+                      body: "Here: I'll start.",
+                      url: "opaque://issue/1/comment/1",
+                      author: makeAuthor("decentralion"),
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+          pullRequests: {
+            pageInfo: {
+              hasNextPage: false,
+              endCursor: "opaque-cursor-prs-v0",
+            },
+            nodes: [
+              {
+                id: "opaque-pr2",
+                title: "Fix typo in README",
+                body: "Surely this deserves much cred.",
+                number: 2,
+                author: makeAuthor("wchargin"),
+                comments: {
+                  pageInfo: {
+                    hasNextPage: false,
+                    endCursor: null,
+                  },
+                  nodes: [],
+                },
+                reviews: {
+                  pageInfo: {
+                    hasNextPage: true,
+                    endCursor: "opaque-cursor-pr2reviews-v0",
+                  },
+                  nodes: [
+                    {
+                      id: "opaque-pr2review1",
+                      body: "You actually introduced a new typo instead.",
```

Comment:
i appreciate the rhyme (cred/instead). i now wish the whole test case were in verse

Reply:
Unfortunately, that would violate the “no fun allowed” rule. ;-)